keyboards/crkbd/keymaps/rpbaptist/keymap.c

```diff
@@ -0,0 +1,254 @@
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
+
+extern uint8_t is_master;
+
+enum layer_names {
+  _COLEMAKDHM,
+  _GAMING,
+  _GAMING_EXT,
+  _NUM,
+  _FN,
+  _NAV,
+  _UTIL
+};
+
+// Base layers
+#define KC_CLMK DF(_COLEMAKDHM)
+#define KC_GAME DF(_GAMING)
+
+// Layer toggle and switch
+#define KC_T_NAV TT(_NAV)
+#define KC_S_NAV MO(_NAV)
+
+#define KC_T_NUM TT(_NUM)
+#define KC_S_NUM MO(_NUM)
+
+// Layer keys with functionality on tap
+#define KC_FNTAB LT(_FN, KC_TAB)
+#define KC_0_NAV LT(_NAV, KC_0)
+#define KC_GMSFT LT(_GAMING_EXT, KC_LSHIFT)
+
+// Tap/mod keys
+#define KC_RCTBR RCTL_T(KC_RBRACKET)
+#define KC_LCTBR LCTL_T(KC_LBRACKET)
+
+#define KC_SFSPC LSFT_T(KC_SPACE)
+#define KC_SFENT LSFT_T(KC_ENTER)
+
+// Global tab forward and backward
+#define KC_TBFWD LCTL(KC_TAB)
+#define KC_TBBCK LCTL(LSFT(KC_TAB))
+
+// Custom key for NUM layer
+#define KC_CTEQL RCTL_T(KC_EQL)
+
+// €
+#define KC_EUR ALGR(KC_5)
+
+#define KC______ KC_TRNS
+#define KC_XXXXX KC_NO
+#define KC_RST   RESET
+#define KC_LTOG  RGB_TOG
+#define KC_LHUI  RGB_HUI
+#define KC_LHUD  RGB_HUD
+#define KC_LSAI  RGB_SAI
+#define KC_LSAD  RGB_SAD
+#define KC_LSPI  RGB_SPI
+#define KC_LSPD  RGB_SPD
+#define KC_LVAI  RGB_VAI
+#define KC_LVAD  RGB_VAD
+
+#define KC_LMOD  RGB_MOD
+
+#define KC_LFMOD  RGB_MODE_FORWARD
+#define KC_LRMOD  RGB_MODE_REVERSE
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_COLEMAKDHM] = LAYOUT_kc( \
+  //,-----------------------------------------.                ,-----------------------------------------.
+        ESC,     Q,     W,     F,     P,     B,                      J,     L,    U,     Y,   SCLN,  BSPC,\
```

Comment:
Layout macros that automatically prepend `KC_` are discouraged, as they do not play well with the Configurator, and make some of the more advanced keycodes such as `MO()` and `LT()` harder to use, since you need to manually define aliases beginning with `KC_` (as you've done above).

Reply:
It's not the default keymap, so it's "okay".  But yeah, we try to discourage the use of them, as they ... tend to be a lot more work for no real gain.

Reply:
It used to be the default keymap for crkbd, which I copied initially. The default has been changed and currently it makes it hard to read. If necessary I can update mine and add some spacing.

Reply:
I updated the keymap.